api/v1beta1/keystoneendpoint.go

```diff
@@ -225,19 +227,119 @@ func GetKeystoneEndpointUrls(
 		return nil, err
 	}
 
+	endpointurls := filterVisibility(ke.Items, visibility)
+
+	return endpointurls, nil
+}
+
+// GetKeystoneEndpointUrlsForServices returns all URLs currently registered. Visibility can be admin, public or internal.
+// If it is nil, all URLs are returned. Optional services parameter can be provided to only return the values for
+// those services matching the `service` (common.AppSelector) label key.
+func GetKeystoneEndpointUrlsForServices(
+	ctx context.Context,
+	h *helper.Helper,
+	namespace string,
+	visibility *string,
+	servicesToKeep []string,
+) ([]string, error) {
+
+	ke, err := GetKeystoneEndpointList(ctx, h, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	keystoneEndpoints := ke.Items
+	if len(servicesToKeep) > 0 && ke != nil {
```

Comment:
ke cannot be nil here as then the previous line would fail to deref nil at `ke.Items`

Reply:
sure, correct